packages/@uppy/companion/src/server/controllers/s3.js

```diff
@@ -36,12 +48,20 @@ module.exports = function s3 (config) {
     const client = req.companion.s3Client
 
     if (!client || typeof config.bucket !== 'string') {
-      res.status(400).json({ error: 'This Companion server does not support uploading to S3' })
+      res.status(400).json({ error: ERROR_BAD_CONFIG })
       return
     }
 
     const metadata = req.query.metadata || {}
     const key = config.getKey(req, req.query.filename, metadata)
+    const bucket = getBucket(req)
+
+    if (!isValidBucket(bucket)) {
+      logger.error(new TypeError(ERROR_BAD_BUCKET))
+      res.status(400).json({ error: ERROR_BAD_CONFIG })
+      return
+    }
```

Comment:
This code is repeated many times. I feel like it could be rewritten to something like this:

```js
function validateBucket(req, res) {
    const bucket = getBucket(req)

    if (isValidBucket(bucket)) return true
    logger.error(ERROR_BAD_BUCKET)
    res.status(400).json({ error: ERROR_BAD_CONFIG })
    return false
}
```

then call:
```js
if (!validateBucket(req, res)) return
```

(also I don't think we need to wrap the log message in a `TypeError`)